### PR TITLE
[fix](txn-insert) Txn insert support ccr

### PR DESCRIPTION
**Pull request by mymeiyi**

### CODE REVIEW

**Comment by dataroaring** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by mymeiyi** [comment]
run p0

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java [hunk 1]
```diff
@@ -1750,11 +1764,29 @@ private boolean commitTxnImpl(TCommitTxnRequest request) throws UserException {
...
+                if (table == null) {
+                    continue;
+                }
+                subTxnIds.add(subTxnInfo.getSubTxnId());
+                subTransactionStates.add(
+                        new SubTransactionState(subTxnInfo.getSubTxnId(), (Table) table,
+                                subTxnInfo.getTabletCommitInfos(), null));
+            }
+            transactionState.setSubTxnIds(subTxnIds);
+            return Env.getCurrentGlobalTransactionMgr().commitAndPublishTransaction(db, request.getTxnId(),
```
Comment (w41ter) [inline fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java:1781]: Is commit info and attachment not required when submitting sub txns?

File: fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java [hunk 2]
```diff
@@ -1265,6 +1267,12 @@ public TBeginTxnResult beginTxn(TBeginTxnRequest request) throws TException {
         try {
             TBeginTxnResult tmpRes = beginTxnImpl(request, clientAddr);
             result.setTxnId(tmpRes.getTxnId()).setDbId(tmpRes.getDbId());
+            if (request.isSetSubTxnNum() && request.getSubTxnNum() > 0) {
+                result.addToSubTxnIds(result.getTxnId());
+                for (int i = 0; i < request.getSubTxnNum() - 1; i++) {
+                    result.addToSubTxnIds(Env.getCurrentGlobalTransactionMgr().getNextTransactionId());
+                }
+            }
```
Comment (w41ter) [inline fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java:1275]: move into `beginTxnImpl`.

File: fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java (hunk 1 above)
Comment (mymeiyi) [reply fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java:1781]: commit info is in SubTransactionState, no attachment for txn insert

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run p0

**Review by w41ter — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**